samza-shell/src/main/bash/run-class.sh

```diff
@@ -62,26 +62,30 @@ if [ -d "$JOB_LIB_DIR" ] && [ "$JOB_LIB_DIR" != "$BASE_LIB_DIR" ]; then
   job_jars=`for file in $JOB_LIB_DIR/*.[jw]ar; do name=\`basename $file\`; if [[ $base_jars != *"$name"* ]]; then echo "$file"; fi; done`
   # get all lib jars and reverse sort it by versions
   all_jars=`for file in $base_jars $job_jars; do echo \`basename $file|sed 's/.*[-]\([0-9]\+\..*\)[jw]ar$/\1/'\` $file; done|sort -t. -k 1,1nr -k 2,2nr -k 3,3nr -k 4,4nr|awk '{print $2}'`
-  # generate the class path based on the sorted result
-  for jar in $all_jars; do CLASSPATH=$CLASSPATH:$jar; done
+  # generate the class path based on the sorted result, all the jars need to be appended on newlines
+  # to ensure java argument length of 72 bytes is not violated
+  for jar in $all_jars; do CLASSPATH=$CLASSPATH"\n $jar "; done
 
   # for debug only
   echo base_jars=$base_jars
   echo job_jars=$job_jars
   echo all_jars=$all_jars
   echo generated combined CLASSPATH=$CLASSPATH
 else
-  # default behaviour
-  # Wildcarding only includes *.jar and *.JAR files in classpath
-  CLASSPATH=$CLASSPATH:"$BASE_LIB_DIR/*";
-  # We handle .war separately
-  for file in $BASE_LIB_DIR/*.war;
+  # default behavior, all the jars need to be appended on newlines
+  # to ensure line argument length of 72 bytes is not violated
+  for file in $BASE_LIB_DIR/*.[jw]ar;
   do
-    CLASSPATH=$CLASSPATH:$file
+    CLASSPATH=$CLASSPATH"\n $file "
   done
   echo generated from BASE_LIB_DIR CLASSPATH=$CLASSPATH
 fi
 
+# Newlines and spaces are intended to ensure proper parsing of manifest in pathing jar
+printf "Class-Path: \n $CLASSPATH \n\n" > manifest.txt
+# Creates a new archive and adds custom manifest information to pathing.jar
+jar -cvmf manifest.txt pathing.jar
```

Comment:
Would be better to create a constant JAR and set it to $JAVA_HOME/bin/jar if JAVA_HOME is defined.

```bash
if [ -z "$JAVA_HOME" ]; then
  JAR="jar"
else
  JAR="$JAVA_HOME/bin/jar"
fi

```

Reply:
Yup I already identified this while testing. Will do